usnistgov/oar-pdr-angular
# Fix/unit test

This checkin migrate unit test from Karma to Jest.

Need some code clean up but it's working.

Here is the instruction on how to run the unit test:

git clone https://github.com/usnistgov/oar-pdr-angular.git
cd oar-pdr-angular/
git submodule update --init --recursive
cd lib
git checkout integration
cd ..
git checkout fix/unit-test

// Install dependencies
npm i --legacy-peer-deps
npm i rxjs@6.6.7 --legacy-peer-deps

npm run build-lib
npm run build-oarlps
npm run build-sw
npm run build-midaslps

Test all:
npm run test

Test single file:
npm run test:midasauthor landingpage.component.spec.ts

Debug with Jest

1.	Place “debugger” to wherever you want to set the breakpoint
2.	Run following command in a console window (the command is from Jest trouble shooting website https://jestjs.io/docs/troubleshooting) in the root folder of a project or library to debug the project or library.

node --inspect-brk ../../node_modules/.bin/jest --runInBand --config ./jest.config.js


To debug a single file:
node --inspect-brk ../../node_modules/.bin/jest –runInBand --config ./jest.config.js filename

See attached instruction on more details migrating to Jest.

[Migrating unit test to Jest.docx](https://github.com/usnistgov/oar-pdr-angular/files/11223383/Migrating.unit.test.to.Jest.docx)
